src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,615 @@
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _summons.DespawnAll();
+        }
+
+        void DoAction(int32 action)
+        {
+            if (action = ACTION_START_EVENT)
+            {
+                TempSummon* sliperryFloor = me->SummonCreature(NPC_SLIPPERY_FLOOR, SummonPositions[7], TEMPSUMMON_MANUAL_DESPAWN);
+                sliperryFloor->SetReactState(REACT_PASSIVE);
+                sliperryFloor->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+                sliperryFloor->AddAura(SPELL_SLIPPERY_FLOOR_AMBIENT, sliperryFloor);
+                for (int counter = 1; counter < 7; counter++)
+                {
+                    if (counter < 4)
+                        me->SummonCreature(NPC_EARTHEN_RING_TOTEM, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                    else
+                        me->SummonCreature(NPC_EARTHEN_RING_FLAMECALLER, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                }
+            }
+        }
+        
+        void UpdateAI(uint32 diff) override
+        {
+            if (say_timer <= diff)
+            {
+                if (Player* player = SelectRandomPlayer(100.0f, false))
+                {
+                    if (count == 0)
+                        sCreatureTextMgr->SendChat(me, SAY_PLAYER_TEXT_1, NULL, CHAT_MSG_SAY, LANG_UNIVERSAL, TEXT_RANGE_NORMAL, 0, TEAM_OTHER, false, player);
+                    if (count == 1)
+                        sCreatureTextMgr->SendChat(me, SAY_PLAYER_TEXT_2, NULL, CHAT_MSG_SAY, LANG_UNIVERSAL, TEXT_RANGE_NORMAL, 0, TEAM_OTHER, false, player);
+                    if (count == 2)
+                        sCreatureTextMgr->SendChat(me, SAY_PLAYER_TEXT_3, NULL, CHAT_MSG_SAY, LANG_UNIVERSAL, TEXT_RANGE_NORMAL, 0, TEAM_OTHER, false, player);
+                    if (count == 3)
+                        TempSummon* ahune = me->SummonCreature(NPC_AHUNE, SummonPositions[0], TEMPSUMMON_DEAD_DESPAWN);
+                    ++count;
+                    say_timer = 3000;
+                }
+            }
+            else
+                say_timer -= diff;
+        }
+
+        Player* SelectRandomPlayer(float range = 0.0f, bool checkLoS = true)
+        {
+            Map* map = me->GetMap();
+            if (!map->IsDungeon())
+                return NULL;
+
+            Map::PlayerList const &PlayerList = map->GetPlayers();
+            if (PlayerList.isEmpty())
+                return NULL;
+
+            std::list<Player*> temp;
+            for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
+                if ((me->IsWithinLOSInMap(i->GetSource()) || !checkLoS) && me->GetVictim() != i->GetSource() &&
+                    me->IsWithinDistInMap(i->GetSource(), range) && i->GetSource()->IsAlive())
+                    temp.push_back(i->GetSource());
+
+            if (!temp.empty())
```

Comment:
Can be replaced by [SelectRandomContainerElement](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/shared/Containers.h#L61)